tutorials/asr/cloud/aws/SageMaker_ASR_Fine-tuning-CommonVoice.ipynb

```diff
@@ -0,0 +1,666 @@
+   "execution_count": null,
+   "id": "ac709398",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# the output path for the model checkpoints/logs\n",
+    "output_path = \"s3://\" + sess.default_bucket() + \"/nemo-output/\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8186678f",
+   "metadata": {},
+   "source": [
+    "We setup the training configuration based on the Esperanto experiments ran [here](https://github.com/andrusenkoau/NeMo/blob/esperanto_example/docs/source/asr/examples/esperanto_asr/esperanto_asr.rst).\n",
+    "\n",
+    "If you change the number of GPUs on the instance, remember to change the values below."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a4b67faa",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "global_batch_size = 1024\n",
+    "n_gpus = 8\n",
+    "accumulate_grad_batches = 8\n",
+    "train_batch_size = int(global_batch_size / n_gpus / accumulate_grad_batches)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4bb61640",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "conf = OmegaConf.load(config_path)\n",
+    "\n",
+    "# data config setup\n",
+    "conf.model.train_ds.batch_size=train_batch_size\n",
+    "conf.model.train_ds.is_tarred = True\n",
+    "conf.model.train_ds.tarred_audio_filepaths = \"/opt/ml/input/data/training/train_tarred_1bk/audio__OP_0..511_CL_.tar\"\n",
+    "conf.model.train_ds.manifest_filepath = \"/opt/ml/input/data/training/train_tarred_1bk/tarred_audio_manifest.json\"\n",
+    "conf.model.validation_ds.manifest_filepath = \"/opt/ml/input/data/testing/manifests/commonvoice_dev_manifest.json.clean.mount\"\n",
+    "\n",
+    "# optimization setup\n",
+    "conf.model.optim.lr = 1e-3\n",
+    "conf.model.optim.sched.name = \"CosineAnnealing\"\n",
+    "\n",
+    "# logging/output setup\n",
+    "conf.exp_manager.exp_dir = \"/opt/ml/model/\"\n",
+    "conf.exp_manager.wandb_logger_kwargs.name = \"cv_sagemaker_8gpus\"\n",
+    "conf.exp_manager.wandb_logger_kwargs.project = \"sagemaker\"\n",
+    "conf.exp_manager.wandb_logger_kwargs.entity = 'nvidia'\n",
+    "conf.exp_manager.create_wandb_logger = True\n",
+    "\n",
+    "# the BPE tokenizer location based on the remote instance mounted directory\n",
+    "conf.model.tokenizer.dir = \"/opt/ml/input/data/training/tokenizers/tokenizer_spe_bpe_v128\"\n",
+    "\n",
+    "# the pre-trained model we want to fine-tune\n",
+    "conf.init_from_nemo_model = \"ssl_en_conformer_large.nemo\"\n",
+    "\n",
+    "# training setup\n",
+    "conf.trainer.accelerator = \"gpu\"\n",
+    "conf.trainer.max_epochs = 1\n",
+    "conf.trainer.accumulate_grad_batches = accumulate_grad_batches\n",
+    "conf.trainer.precision = 16\n",
+    "\n",
+    "# uncomment if you want to use W&B, you'll need to pass your token at a later step\n",
+    "conf.exp_manager.wandb_logger_kwargs.name = \"common_voice_esperanto\"\n",
```

Comment:
Add resume flags. It's harmless if it completes the job, but if it fails then at least pointing to same paths with checkpoints will allow continuation